_rules/SC2-4-6-descriptive-labels.md

```diff
@@ -21,27 +21,39 @@ authors:
+- inherits from the [abstract](https://www.w3.org/TR/wai-aria/#abstract_roles) `input` or `select` role, and 
 - does not have a [required context](https://www.w3.org/TR/wai-aria/#scope) role that itself inherits from one of those roles.
-- The `option` role is not part of the list of applicable roles, because it does not meet the definition of a [User interface component](https://www.w3.org/TR/WCAG21/#dfn-user-interface-components). This means [WCAG 2.1](https://www.w3.org/TR/WCAG21/) does not require it to have an accessible name.
 
-**Note**: This rule is a partial check for WCAG 2.1 success criterion 2.4.6, which applies to all labels. "Label" is used in its general sense and includes text or other components with a text alternative that is presented to a user to identify a component within Web content.
+**Note:**
+- The `option` role is not part of the list of applicable roles, because it has a required context role that inherits from the `select` role. Furthermore, `option` does not meet the definition of a [User interface component](https://www.w3.org/TR/WCAG21/#dfn-user-interface-components). This means [WCAG 2.1](https://www.w3.org/TR/WCAG21/) does not require it to have an accessible name.
+
+**Note**: This rule is a partial check for WCAG 2.1 success criterion 2.4.6, which applies to all labels. "Label" in WCAG is used in its general sense and includes text or other components with a text alternative that is presented to a user to identify a component within Web content.
 
-### Expectation
+### Expectation 1
 
-Each target element describes the purpose of the associated form field element.
+The `label` element, if there is any, together with its context, describe the purpose of the associated form field element.
+
+**Note:** Context in this case could be e.g. headings, fieldsets and legends, text that is located close by etc.
+
+**Note**: Labels do not need to be lengthy. A word, or even a single character, may suffice.
+
+### Expectation 2
+
+The [accessible name](#accessible-name), if there is any, together with it's context, describe the purpose of the associated form field element.
+
+**Note:** Context in this case could be e.g. headings, fieldsets and legends, text that is located close by etc.
 
 **Note**: Labels do not need to be lengthy. A word, or even a single character, may suffice.
 
 ## Assumptions
 
-_There are currently no assumptions._
+This rule assumes that while having an important context that is not [programatically determinable](#https://www.w3.org/TR/WCAG21/#dfn-programmatically-determinable) might be a violation under other [WCAG](https://www.w3.org/TR/WCAG21/) success criteria (e.g. 1.3.1 Info and Relationships, if text that has not been marked up as headings is used to split a form into different sections), this is allowed under success criterion 2.4.6 Headings and Labels: Headings and labels describe topic or purpose.
```

Comment:
@ShadowBB, seems that there is a disagreement on your WCAG interpretation.

Reply:
Hmmm. Failing example 6 feels really wrong to me for some reason. I would say that the labels still describe the topic/purpose even without(!) their heading context. Especially because if these two headings (and their content) would be on different pages I would pass them in a heartbeat.

But I do get your point. Because currently "fill in" would also pass this rule as long as it has a heading "first name":

```html
<h1>First name</h1>
<label for=FirstName>Fill in</label>
<input type="text" id="FirstName">
<h1>Last name</h1>
<label for=LastName>Fill in</label>
<input type="text" id="Lastname">
```

Which I would definitely fail.

So I think Carlos is correct in mentioning the fact that there is something wrong with this assumption. 

I think the context is only allowed to differentiate the topic/purpose of the field from others like it.

So the expectation 1 would then read:

"The `label` element, if there is any describe the purpose of the associated form field element. The context of the element can be used to differentiate the purpose from other fields on the same page."

And the note should then read:

**Note:** In this case, context can be created by headings, fieldsets and legends, text that is near the control, etc. Differentiation by context is usually only needed if there are two fields with similar purpose on the same page. For example if the same page asks for both a shipping and billing address.

Expectation 2 should be handled similarly.

Reply:
@carlosapaduarte: Would you agree with this solution?

Reply:
@ShadowBB Yes, I do agree with your proposal. Labels won't need context to be meaningful (thus failing the "Fill in" example), but one can use context to distinguish between meaningful labels. Super!

Just a minor edit in the text for the expectation: 
"The label element, if there is any, describes the purpose of the associated form field element. The context of the element can be used to differentiate the purpose from other fields on the same page."

Reply:
@carlosapaduarte Awesome. And I agree with your minor edit.

Reply:
Thanks for the suggestion! 

I have changed both expectations to match, edited the Passed test case description for the shipping/billing address one and added a failed test case with "fill in"´+ a meaningful context.